src/lib.rs

```diff
@@ -469,6 +480,22 @@ impl<N, E, Ix = DefIndex> Dag<N, E, Ix> where Ix: IndexType {
 }
 
 
+/// After adding a new edge to the graph, we use this function immediately after to check whether
+/// or not we need to check for a cycle.
+/// 
+/// If our parent *a* has no parents or our child *b* has no children, or if there was already an
+/// edge connecting *a* to *b*, we know that adding this edge has not caused the graph to cycle.
+fn must_check_for_cycle<N, E, Ix>(dag: &Dag<N, E, Ix>,
+                                  a: NodeIndex<Ix>,
+                                  b: NodeIndex<Ix>,
+                                  new_edge: EdgeIndex<Ix>) -> bool
+    where Ix: IndexType,
+{
+    dag.parents(a).next(dag).is_some() && dag.children(b).next(dag).is_some()
+    && !dag.children(a).any(dag, |_, e, _| e != new_edge)
```

Comment:
I think you mean to check that there is an edge from a to b here, needs a condition that the child is b?